Newsletter January 2022 (#310)
ropensci/roweb3: content/blog/2022-01-10-ropensci-news-digest-january-2022/index.Rmd

### potools: Translate your package user-facing communications like messages, warnings, and errors

Your package might be used by people at ease in different languages than the one you initially wrote e.g. error messages in.
The [potools package](https://michaelchirico.github.io/potools/index.html) maintained by [Michael Chirico](https://github.com/MichaelChirico) provides "Tools for Portability and Internationalization of R packages".
It features great vignettes including one for [package developers](https://michaelchirico.github.io/potools/articles/developers.html) and one for [package creators](https://michaelchirico.github.io/potools/articles/translators.html).
The potools package is used in particular in the pkgdown package ([user-facing docs for the language option](https://pkgdown.r-lib.org/articles/pkgdown.html?q=language#language).

### How to provide tibbles only to those who like them

"Only use tibbles if tibble installed": this sounds great, doesn't it?
It's the title of a [PR by Hadley Wickham](https://github.com/allisonhorst/palmerpenguins/pull/25) to the palmerpenguins package maintained by Allison Horst.

> Key to this PR is the used of `delayedAssign()` in `data/penguins.R`.

Reviewer: "is the used of" -> "is the use of"

    stringr::str_pad(lubridate::month(rmarkdown::yaml_front_matter(knitr::current_input())$date), 2, "0", side = "left"),
    stringr::str_pad(lubridate::day(rmarkdown::yaml_front_matter(knitr::current_input())$date), 2, "0", side = "left"),
    rmarkdown::yaml_front_matter(knitr::current_input())$slug
    )
english <- function(x) {
  as.character(english::english(x))
}

nice_string <- function(...) {
  glue::glue_collapse(..., sep = ", ", last = ", and ")
}
```
<!-- Before sending DELETE THE INDEX_CACHE and re-knit! -->

Dear rOpenSci friends, it's time for our monthly news roundup!
<!-- blabla -->
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

<!-- to be curated manually -->

### Co-working events
<!-- Can use same coworking text every month and ping Steffi to ask if there are any special guests to add -->
Join us for social coworking & office hours monthly on 1st Tuesdays! Hosted by Steffi LaZerte and Nick Tierney. Everyone welcome. No RSVP needed. Consult our [Events](/events) page to find your local time and how to join.

Our next sessions are:

> * [Monday, 31 January 2022 17:00 PST](/events/coworking-2022-02/) (9 AM Australian Western);

Reviewer 2: Just because the actual date is different, a bit messy, but I think necessary to be clear!
```suggestion
* [Monday, 31 January 2022 17:00 PST](/events/coworking-2022-02/) (Tuesday, Feb 1st 09:00 Australian Western);
```

* [Tuesday, 01 March 2022 09:00 PST](/events/coworking-2022-03/) (17:00 UTC).

Find out about more [events](/events).

> ### Community Management Transition for rOpenSci

Reviewer 2: Consistent with Sentence case headings elsewhere (assuming Community Management is a title :grin:)
```suggestion
### Community Management transition for rOpenSci
```

The fantastic Stefanie Butland has moved on from her position as rOpenSci's community manager.
Her [blog post](/blog/2022/01/14/msg-from-stefanie/) summarized the incredible work she's done over the last 5.5 years.

> As to rOpenSci future new community manager, it might be you: keep an eye out for a job posting in February and even before that, email info@ropensci.org if you think you might be interested.

Reviewer 2: Apparently I needed more coffee when I first read this, sorry, I missed a couple of things!
Also, the text here is slightly different from the index.md (i.e. the placement of "that, " vs. ", that"). I assume this is correct, and it'll be compiled to md later?
```suggestion
As for rOpenSci's future new community manager, it might be you: keep an eye out for a job posting in February and even before that, email info@ropensci.org if you think you might be interested.
```

### An update on R-universe

> We are making steady progress on the development of [r-universe](https://r-universe.dev), rOpenSci’s open platform for publishing and discovering R packages and other research material. Currently r-universe is tracking 3548 R packages and 3602 [articles](https://r-universe.dev/articles/) from 851 [maintainers](https://r-universe.dev/maintainers/) in 384 [cranlike repositories](https://r-universe.dev/organizations/). Over the past months we have added several new API's and front-end features, including [support for badges](/blog/2021/10/14/runiverse-badges/), [rss feeds](/blog/2021/11/24/runiverse-badges/), [orcid links](https://twitter.com/opencpu/status/1468323209770582018), and [support for renv](/blog/2022/01/06/runiverse-renv/) to restore packages from r-universe for reproducibility purposes.

Reviewer 2: I *think* it's supposed to be capital R?

```suggestion
We are making steady progress on the development of [R-universe](https://r-universe.dev), rOpenSci’s open platform for publishing and discovering R packages and other research material. Currently R-universe is tracking 3548 R packages and 3602 [articles](https://r-universe.dev/articles/) from 851 [maintainers](https://r-universe.dev/maintainers/) in 384 [cranlike repositories](https://r-universe.dev/organizations/). Over the past months we have added several new API's and front-end features, including [support for badges](/blog/2021/10/14/runiverse-badges/), [rss feeds](/blog/2021/11/24/runiverse-badges/), [orcid links](https://twitter.com/opencpu/status/1468323209770582018), and [support for renv](/blog/2022/01/06/runiverse-renv/) to restore packages from R-universe for reproducibility purposes.
```

> Under the hood we made a [fundamental change](https://twitter.com/opencpu/status/1468323209770582018) to give maintainers a more central role in r-universe: the dashboards and APIs for [`https://{user}.r-universe.dev`](https://hadley.r-universe.dev) now also include packages and articles from other organizations that are maintained by that user, while cross-linking maintainers and organizations. This should make it easier to browse content help you discover interesting researchers and material.

Reviewer 2: ```suggestion
Under the hood we made a [fundamental change](https://twitter.com/opencpu/status/1468323209770582018) to give maintainers a more central role in R-universe: the dashboards and APIs for [`https://{user}.r-universe.dev`](https://hadley.r-universe.dev) now also include packages and articles from other organizations that are maintained by that user, while cross-linking maintainers and organizations. This should make it easier to browse content help you discover interesting researchers and material.
```

)

registry <- registry %>%
  dplyr::rowwise() %>%
  dplyr::mutate(
  owner = strsplit(urltools::path(url), "/")[[1]][1],
  repo = strsplit(urltools::path(url), "/")[[1]][2]
) %>%
  dplyr::filter(
    !is.na(owner)
  )
packages <- split(registry, seq(nrow(registry)))
get_release <- function(repo) {
  info <- gh::gh(
    "GET /repos/{owner}/{repo}/releases",
    owner = repo$owner,
    repo = repo$repo,
    per_page = 3,
    description = repo$description
  )
  info <- info[!purrr::map_lgl(info, "draft")]
  if(length(info) == 0 || anytime::anytime(info[[1]]$published_at) < last_newsletter) {
    return(NULL)
  }
  
  tibble::tibble(
    package = repo$package,
    version = info[[1]]$tag_name,
    url = info[[1]]$html_url,
    description = repo$description
  )
}
releases <- purrr::map_df(
  packages,
  get_release
)
releases <- split(releases, seq(nrow(releases)))
format_release <- function(release) {
  sprintf(
    '[%s](https://docs.ropensci.org/%s "%s") ([`%s`](%s))',
    release$package,
    release$package,
    release$description,
    release$version,
    release$url
  )
}
all_releases <- purrr::map_chr(releases, format_release)
text <- nice_string(all_releases)
```

> The following `r if (length(releases) > 1) english(length(releases))` package`r if (length(releases) > 1) "s"` `r if (length(releases) > 1) "have" else "has"` had an update since the latest newsletter: `r text`.

Reviewer 2: ```suggestion
The following `r if (length(releases) > 1) english(length(releases))` package`r if (length(releases) > 1) "s"` `r if (length(releases) > 1) "have" else "has"` had an update since the last newsletter: `r text`.
```

> ## Software Peer Review

Reviewer 2: Package canaper appears twice in the list under 3/reviewer assigned. Perhaps need a unique()/distinct()?

Author: Oh good catch! Turns out the submission had two progress labels https://github.com/ropensci/software-review/issues/475#event-5912273678

      purrr::map_chr(tech_notes, format_post),
      collapse = "\n\n"
    )
  )
  cat("\n\n")
}
```

## Use cases

```{r usecases}
# rerun get_use_cases.R at the same time
usecases <- jsonlite::read_json("../../../data/usecases/usecases.json")
get_one_case <- function(usecase) {
  tibble::tibble(
    title = usecase$title,
    reporter = usecase$reporter,
    url = usecase$url,
    image = usecase$image,
    date = anytime::anydate(usecase$date)
  )
}
usecases <- purrr::map_df(usecases, get_one_case)
usecases <- dplyr::filter(usecases, date >= as.Date(last_newsletter))
usecases <- split(usecases, seq(nrow(usecases)))
```

`r snakecase::to_sentence_case(english(length(usecases)))` use cases of our packages and resources have been reported since we sent the last newsletter.

```{r usecases2, results='asis'}
format_case <- function(usecase) {
  string <- sprintf("* [%s](%s). Reported by %s.", sub("\\.$", "", usecase$title), usecase$url, usecase$reporter)
}
cat(
  paste0(
    purrr::map_chr(usecases, format_case),
    collapse = "\n\n"
  )
)
```

Explore [other use cases](/usecases) and [report your own](https://discuss.ropensci.org/c/usecases/10)!

## Call for maintainers

> There's no open call for new maintainers at this point but you can refer to our [contributing guide](https://contributing.ropensci.org/) for finding ways to get involved!

Reviewer 2: ```suggestion
There are no open calls for new maintainers at this point but you can refer to our [contributing guide](https://contributing.ropensci.org/) for finding ways to get involved!
```

* Add a script `.aspell/defaults.R` that will refer to both the standard dictionary ("en_stats") and the one you just created ("coolstuff").

```r
Rd_files <- vignettes <- R_files <- description <-
    list(encoding = "UTF-8",
         language = "en",
         dictionaries = c("en_stats", "coolstuff"))
```

Voilà, goodbye quotes in `DESCRIPTION`!
Thanks to [Jim Hester and Kirill Müller](https://github.com/cynkra/fledge/pull/171#discussion_r762045958) for this cool trick that's not [used a lot](https://github.com/search?q=user%3Acran+extension%3Ards+path%3A.aspell&type=Code&l=&l=) (yet?).
It _is_ documented in R, see the sentence beginning with "Suitable values for" in https://stat.ethz.ch/R-manual/R-patched/library/utils/html/aspell-utils.html

### potools: Translate your package user-facing communications like messages, warnings, and errors

Your package might be used by people at ease in different languages than the one you initially wrote e.g. error messages in.
The [potools package](https://michaelchirico.github.io/potools/index.html) maintained by [Michael Chirico](https://github.com/MichaelChirico) provides "Tools for Portability and Internationalization of R packages".

> It features great vignettes including one for [package developers](https://michaelchirico.github.io/potools/articles/developers.html) and one for [package creators](https://michaelchirico.github.io/potools/articles/translators.html).

Reviewer 2: ```suggestion
It features great vignettes including one for [package developers](https://michaelchirico.github.io/potools/articles/developers.html) and one for [package translators](https://michaelchirico.github.io/potools/articles/translators.html).
```
